src/pynxtools/nomad/schema.py

```diff
@@ -191,25 +191,43 @@ def normalize(self, archive, logger):
         ]
         if not identifiers:
             return
-        self.AnchoredReferences = []
+        self.NXidentifiers = []
         for identifier in identifiers:
             if not (val := getattr(self, identifier)):
                 continue
-            # identifier_path = f"{self.m_def.name}_{identifier.split('__field')[0]}"
-            field_n = identifier.split("__field")[0]
-            logger.info(f"Lab id {val} to be created")
-            nx_id = AnchoredReference(lab_id=val, name=field_n)
-            nx_id.m_set_section_attribute(
-                "m_nx_data_path",
-                self.m_get_quantity_attribute(identifier, "m_nx_data_path"),
-            )
-            nx_id.m_set_section_attribute(
-                "m_nx_data_file",
-                self.m_get_quantity_attribute(identifier, "m_nx_data_file"),
-            )
+            if isinstance(val, dict):
+                for idname, idobj in val.items():
+                    # identifier_path = f"{self.m_def.name}_{identifier.split('__field')[0]}"
+                    field_n = idname.split("__field")[0]
+                    logger.info(f"Lab id {idobj.value} to be created")
+                    nx_id = AnchoredReference(lab_id=idobj.value, name=field_n)
+                    nx_id.m_set_section_attribute(
+                        "m_nx_data_path",
+                        self.m_get_quantity_attribute(idname, "m_nx_data_path"),
+                    )
+                    nx_id.m_set_section_attribute(
+                        "m_nx_data_file",
+                        self.m_get_quantity_attribute(idname, "m_nx_data_file"),
+                    )
```

Comment:
In case the `val` is a dict, this code may not work. In this case `self` will  have any attribute with the name `idname`.